src/audio/buffer.c

```diff
@@ -56,21 +56,46 @@ struct comp_buffer *buffer_new(struct sof_ipc_buffer *desc)
 	assert(!memcpy_s(&buffer->ipc_buffer, sizeof(buffer->ipc_buffer),
 		       desc, sizeof(*desc)));
 
-	buffer->size = desc->size;
-	buffer->alloc_size = desc->size;
-	buffer->w_ptr = buffer->addr;
-	buffer->r_ptr = buffer->addr;
-	buffer->end_addr = buffer->addr + buffer->ipc_buffer.size;
-	buffer->free = buffer->ipc_buffer.size;
-	buffer->avail = 0;
-
-	buffer_zero(buffer);
+	buffer_init(buffer, desc->size);
 
 	spinlock_init(&buffer->lock);
 
 	return buffer;
 }
 
+int buffer_resize(struct comp_buffer *buffer, uint32_t size)
```

Comment:
We already have buffer_set_size and now we are adding buffer_resize.
I would keep just one of the functions to no confuse any future API users.

Reply:
so you mean incorporating the realloc to buffer_set_size?

Reply:
@tlauda you mean something like this updated version? So buffer_set_size is now using realloc and if realloc is working correctly it should not do anything with trying to change with same original size. So semantics of buffer_set_size would change following way: previously we would set size smaller even if bigger chuck is reserved (upscaling not possible), now we can change up or down and reserved size should be the same...?